docs(agents): add dd-apm-sdk-review pre-push review skill (#9606)
DataDog/dd-trace-js: .agents/skills/dd-apm-sdk-review/reviewers/conventions.md

- `.agents/skills/apm-integrations/SKILL.md` (+ `references/`) — instrumentation/plugin architecture and rules.
- `.agents/skills/llmobs-integration/SKILL.md`, `.agents/skills/llmobs-testing/SKILL.md`, `.agents/skills/serverless-integrations/SKILL.md`.
- `docs/API.md` (public option/plugin docs), `index.d.ts` / `index.d.v5.ts` (public types), `docs/test.ts` (type smoke test).
- `eslint.config.mjs` + `eslint-rules/` — mechanically enforced style.
- `.github/pull_request_template.md`, `.github/workflows/pr-title.yml` (PR title/label contract).

**Read those files as part of this review.** They are the specification you are
reviewing against. If a rule there contradicts your instinct, the rule wins.
Quote the rule you're invoking when you report a finding.

## Mechanical checks — run these, don't eyeball them

Read **AGENTS.md § "Code Style & Linting"** for the style rules and the lint entry points, then run them against the changed files.

```bash
npm run lint        # check_licenses + check-no-* + eslint --max-warnings 0 + codeowners audit + verify-exercised-tests

> npm run lint:fix    # eslint --fix + the check-* scripts ONLY - does NOT run

Reviewer: wouldn't this break the read only rule in _common.md

Author: I think the first one would take precedent, but either way this doesn't belong here. I'll fix that.

Author: Fixed — the reviewer commands are check-only now, and `_common.md`'s read-only rule explicitly covers formatters and `--fix` variants.

Good catch beyond this line: auditing the other SDK repos turned up the same thing queued in 8 of 9 (`spotlessApply`, `standard:fix`, `composer fix-lint`, `cargo fmt`, etc.), so this would have shipped to java/py/rb/php too. There's now a check that fails the build if a reviewer prompt invokes a mutating command.

**Read those files as part of this review.** They are the specification you are
reviewing against. If a rule there contradicts your instinct, the rule wins.
Quote the rule you're invoking when you report a finding.

## Mechanical checks — run these, don't eyeball them

Check-mode only. Anything below that would rewrite files is the author's to run,
not yours; if a check fails, report it.

Read **AGENTS.md § "Code Style & Linting"** for the style rules and the lint entry points, then run them against the changed files.

```bash
npm run lint        # check_licenses + check-no-* + eslint --max-warnings 0 + codeowners audit + verify-exercised-tests
npm run type:check  # tsc --noEmit -p tsconfig.dev.json
./node_modules/.bin/eslint --max-warnings 0 <changed files>
```

Generated-artifact verifiers CI runs, which a config or plugin change can break:
```bash
npm run verify:config:types            # scripts/generate-config-types.js --check
npm run verify:supported-integrations
npm run verify:electron-package
npm run verify:workflow-job-names
npm run test:eslint-rules              # eslint-rules/*.test.mjs

> ```

Reviewer 2: Not necessarily wrong to have here, but also this should have been done as part of a different agentic flow, e.g. pre-commit hooks that require running the linter

Author: Agreed; leaving it until a hook exists, since it's the only signal today. All check-only now after Pablo's catch.

  manifest, CODEOWNERS) — is that registration present? Missing wiring means the
  code silently never runs, which is Blocking.
- **Commit and PR hygiene** as this repo requires:

Read **AGENTS.md § "Pull Requests and CI"** and its subsections "Commit Messages", "PR Requirements", and "Flaky tests", plus **§ "Always Consider Backportability"**. Those own the title format and allowed types, the semver label rules, the PR template, the all-green policy, the flaky-test policy, and the `DD_MAJOR` guard for breaking changes.

Only the parts not stated there:
- The authoritative list of allowed title types is `.github/workflows/pr-title.yml` (`PR_TITLE_PATTERN`), which also auto-syncs the type/scope/semver labels from the title - so a wrong title produces a wrong release label.
- `only-land-on-next` is applied by hand for changes that must not land on stable release lines (CONTRIBUTING.md, "Indicate intended release targets").
- There is no changelog file in this repo: the PR title is the release note. Audit the title and labels rather than asking for a changelog entry.
- No in-repo rule mandates `gh --repo` flags or a fork-vs-branch policy; do not invent one.

## Do not

> - Do not invent conventions the repo doesn't state and doesn't practice.

Reviewer 2: ```suggestion
- Do not invent conventions that do not exist in the repo
```

Author: Taken.
